docs/ios/6x/automatic-instrumentation/hang-detection.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Hang Detection
+description: Automatically detect and monitor main thread hangs in your iOS app with Embrace
+sidebar_position: 7
+---
+
+# Hang Detection
+
+The Embrace SDK automatically monitors your application's main thread for app hangs, providing visibility into UI freezes and unresponsive behavior that can frustrate users.
+
+## What are Hangs?
+
+A hang occurs when your app's main thread (UI thread) is blocked for too long, preventing the app from responding to user interactions. During a hang, your app may appear frozen—buttons don't respond, animations stop, and the UI becomes unresponsive.
+
+Common causes of hangs include:
+
+- Performing heavy computations on the main thread
+- Synchronous network requests
+- Large file I/O operations
+- Inefficient database queries
+- Deadlocks or race conditions
+- Slow third-party SDK initialization
+
+Even hangs as short as 250 milliseconds can be noticeable to users and negatively impact the user experience.
+
+For more information about understanding and improving hangs in iOS apps, see Apple's documentation:
+
+- [Understanding Hangs in Your App](https://developer.apple.com/documentation/xcode/understanding-hangs-in-your-app)
+- [Improving App Responsiveness](https://developer.apple.com/documentation/xcode/improving-app-responsiveness)
+
+## How Hang Detection Works
+
+The SDK automatically monitors the main thread using a dedicated watchdog thread. When the main thread is blocked for longer than 249 milliseconds (Apple's recommended threshold), a hang is detected and reported with:
+
+- Duration of the hang
+- Stack traces to identify the blocking code (when enabled)
+- Associated session and user context
+
+## Key Benefits
+
+- Automatically detect UI freezes and unresponsive behavior
+- Identify code causing main thread blockages
+- Track hang frequency and duration
+- No code changes required
+
+## Configuration
+
+Hang detection is enabled by default. You can customize settings during development or testing by implementing a custom `EmbraceConfigurable`:
+
+```swift
+import EmbraceIO
+import EmbraceConfiguration
+
+// Create a custom configuration class
+class CustomConfig: EmbraceConfigurable {
+    // Customize hang detection limits
+    var hangLimits = HangLimits(
+        hangPerSession: 200,    // Max hangs to capture per session
+        samplesPerHang: 5       // Max stack trace samples per hang
+    )
+
+    // Required EmbraceConfigurable properties with defaults
+    var isSDKEnabled: Bool = true
```

Comment:
you don't need to cover all of these, maybe just add a "// ... add all other configs here" or something.